auth_totp/models/res_users.py

```diff
@@ -50,53 +56,38 @@ def _check_enabled_with_authenticator(self):
 
     @api.model
     def check_credentials(self, password):
-        try:
+        """Add MFA logic to core authentication process
+
+        Overview:
+            * If user does not have MFA enabled, defer to parent logic
+            * If user has MFA enabled and has gone through MFA login process
+              this session or has correct device cookie, defer to parent logic
+            * If neither of these is true, call parent logic. If successful,
+              prevent auth while updating session to indicate that MFA login
+              process can now commence
+        """
+        user_model_sudo = self.sudo()
+        user = user_model_sudo.search([('id', '=', self.env.uid)])
```

Comment:
`user = self.sudo().browse(self.env.uid)`

Reply:
Why not `self.env.user.sudo()`?

Reply:
Actually, it turns out that ``env.user`` already [bypasses access restrictions](https://github.com/odoo/odoo/blob/10.0/odoo/api.py#L796), so I'll just drop these lines entirely and use that as needed. Thanks for the push in the right direction.